lib/OpenQA/Utils.pm

```diff
@@ -1004,19 +1010,29 @@ sub read_test_modules {
                     log_debug("Cannot read file: $file");
                 }
             }
+
+            if (($step->{is_parser_text_result} = $source && $source eq 'parser' && $text && $step->{text_data})) {
```

Comment:
Can you express this condition better? you're basically assigning and also as a bonus set `$step->{is_parser_text_result}` on a separate line :)

Reply:
I did the usual double parenthesis so that it is just `=` and not `==` shouldn't cause confusion. And isn't the meaning clear? The variable name describes it. I also made it like this to avoid an extra variable or accessing the hash value again.

Reply:
@Martchus I don't understand how golf works...  :D

```
$step->{is_parser_text_result} = 0;
if ($source && $source eq 'parser' && $text && $step->{text_data}) {
   $step->{is_parser_text_result} = 1;
```

Reply:
Changed, it is now "readable".

Reply:
:rofl: